### DISCUSSION TITLE
api.Div leading to a float

### DISCUSSION THREAD

**Discussion by infinite-pursuits** [Consensys/gnark#709] [Q&A]
Hi,
I do ```api.Div``` in my circuit. The result of this division should be a float and it is represented as a big number in the circuit. How do I interpret this big number? Like for negative integers,  -1 gets converted to P-1. Looking for a similar way to interpret.

**Comment by ivokub** [comment]
See https://github.com/ConsenSys/gnark/discussions/711#discussioncomment-6047454. In short, gnark does not support floating-point arithmetic. I do recommend adapting the algorithms such that you wouldn't have to perform floating-point operations. For example, instead of L2 metric you could use L1 metric.

**Reply by infinite-pursuits** [reply]
Yes that makes sense.
I wrote a hint function to calculate the quotient and the remainder and verified that in the circuit. and also squared the L2 norm. Cant use L1 unfortunately.

**Reply by sid-alluri** [reply]
Hi @infinite-pursuits 

Are you able to perform the division?
